docs/architecture/adr-112-proposer-based-timestamps.md

```diff
@@ -9,40 +9,40 @@
+Each validator adds a timestamp to each `Precommit` message it sends.
+The timestamp a correct validator sends is either the validator's current known Unix time or one millisecond greater than the previous block time, depending on which value is greater.
 When a block is produced, the proposer chooses the block timestamp as the weighted median of the times in all of the `Precommit` messages the proposer received.
-The weighting is proportional to the amount of voting power, or stake, a validator has on the network.
-This mechanism for producing timestamps is both deterministic and byzantine fault tolerant.
+The weighting is defined by the amount of voting power, or stake, each validator has on the network.
+This mechanism for producing timestamps is both deterministic and Byzantine fault tolerant.
 
 This current mechanism for producing timestamps has a few drawbacks.
 Validators do not have to agree at all on how close the selected block timestamp is to their own currently known Unix time.
-Additionally, any amount of voting power `>1/3` may directly control the block timestamp.
+Additionally, any amount of voting power `>1/3` may control the block timestamp.
 As a result, it is quite possible that the timestamp is not particularly meaningful.
 
-These drawbacks present issues in the CometBFT protocol.
+These drawbacks present issues in CometBFT.
 Timestamps are used by light clients to verify blocks.
-Light clients rely on correspondence between their own currently known Unix time and the block timestamp to verify blocks they see;
-However, their currently known Unix time may be greatly divergent from the block timestamp as a result of the limitations of `BFTTime`.
+Light clients rely on correspondence between their own currently known Unix time and the block timestamp to verify blocks they see.
+However, their currently known Unix time may be greatly divergent from the block timestamp as a result of the limitations of `BFT Time`.
 
-The proposer-based timestamps specification suggests an alternative approach for producing block timestamps that remedies these issues.
+The [Proposer-Based Timestamps specification (PBTS)][pbts-spec] suggests an alternative approach for producing block timestamps that remedies these issues.
 Proposer-based timestamps alter the current mechanism for producing block timestamps in two main ways:
 
-1. The block proposer is amended to offer up its currently known Unix time as the timestamp for the next block instead of the `BFTTime`.
+1. The block proposer is amended to offer up its currently known Unix time as the timestamp for the next block instead of the `BFT Time`.
 1. Correct validators only approve the proposed block timestamp if it is close enough to their own currently known Unix time.
 
 The result of these changes is a more meaningful timestamp that cannot be controlled by `<= 2/3` of the validator voting power.
```

Comment:
In when sense is it better than BFT Time in this aspect?
My understanding:
* In BFT Time: Byzantine validators (say, close to 1/3 of voting power) can affect the time by timestamping their votes with nonsensical values. They will affect the median value, but the median value will still be that from a correct validator
* In PBTS: Byzantine proposers also have some degree of "freedom" to propose values that deviate from the real value of their clock, as long as `timely` passes in 2/3 of validators
So, in both cases, byzantine validators have a limited influence on the final timestamp value, albeit, via different mechanisms.

Reply:
The point here, and this is some analysis from PBTS v1 (namely, I was not part of it): https://github.com/cometbft/cometbft/blob/main/spec/consensus/proposer-based-timestamp/v1/pbts_001_draft.md

- BF Time: If more than 1/2 of the voting power is held by faulty validators, then the time is under total control of the faulty validators. Here faulty combines Byzantine and "correct"  validators with non synchronized clocks.
- PBTS: the block time can be corrupted only in the extreme case when >2/3 of the validators are faulty. Corrupted here means outside the synchrony window.

Reply:
Yeah, but is you allow "correct" process to have unsynchronized clocks on one side and forbid is on the other side, aren't we then comparing apples and oranges?

Reply:
Yes, we are. But we have also changed the system model.

I updated point 2 there with the assumption of synchronized clocks.